mantidproject/mantid
# Serialise the spectrum viewer

**This should not be merged before the PR for #16961**

This PR allows the spectrum viewer to serialise it's state to a Mantid project file

**To test:**
Saving to a Mantid project with a spectrum viewer window open should now save the state of the spectrum viewer to file as well. Best way to test is to play around and ensure all the options are correctly saved and reapplied. Some specific things to test:
- Multiple workspaces open on the interface
- Min/max range
- Slider values
- Colormaps (both from file and preexisting)
- EMode/Fixed
- Selected points on the spectrogram and vertical/horizontal 1D plots.

See if you can spot anything I've missed.

Fixes #16963.

For release notes see issue #17109.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
